src/winmain.cpp

```diff
@@ -917,6 +917,43 @@ bool runInstaller(const wstring& app2runPath, const wstring& binWindowsClassName
 	return true;
 }
 
+// Returns a folder suitable for exe installer download destination.
+std::wstring getDestDir()
+{
+	// Some users might unset the temp variable, so need to account for that.
+	// Other fallbacks: %AppData%, %UserProfile% or GetModuleFileName().
+	size_t requiredSize = 0;
+	std::vector<wchar_t> envPath(MAX_PATH + 1, 0);
+	for (const wchar_t* env : { L"TEMP", L"TMP" })
+	{
+		errno_t e = _wgetenv_s(&requiredSize, envPath.data(), envPath.size(), env);
+		if (requiredSize == 0)
+		{
+			// Variable doesn't exist.
+			continue;
+		}
+		else if (e == ERANGE)
+		{
+			// Buffer is too small.
+			envPath.resize(requiredSize);
+			e = _wgetenv_s(&requiredSize, envPath.data(), envPath.size(), env);
+		}
+		if (e == 0)
+		{
+			// Trim and stop.
+			envPath.resize(requiredSize);
+			break;
+		}
+		else
+		{
+			envPath.clear();
+		}
+	}
+	if (!envPath.empty() && envPath[0])
+		return std::wstring(envPath.begin(), envPath.end());
+	return L".";

```

Comment:
I can reproduce the crash by setting `%TMP%` and `%TEMP%` to an empty string.
However, this PR doesn't fix the crash when Notepad++ is installed into it's default destination `C:\Program Files`.

I replace the line 
```
return L".";
```
by
```
return L"c\\blabla\\";
```
and create `blabla` folder before launching the update, then it works.

Obviously `Program Files` folder needs Admin privilege but GUP.exe has none.

Reply:
@donho What would you prefer as a fallback? How about `%AppData%`? Or even `%AppData%\Roaming\Notepad++\`?

Reply:
%AppData%\Roaming\Notepad++\ will work only if it does exist.
I would say `%USERPROFILE%\AppData\Local\Temp` - if it doesn't exist, just create it.

Reply:
Done.